tests/unit/api/objectReplicationMD.js

```diff
@@ -0,0 +1,295 @@
+    });
+
+    afterEach(() => cleanup());
+
+    it('should not update object metadata', done =>
+        putObjectAndCheckMD(keyA, emptyReplicationMD, done));
+
+    it('should not update object metadata if putting object ACL', done =>
+        async.series([
+            next => putObjectAndCheckMD(keyA, emptyReplicationMD, next),
+            next => objectPutACL(authInfo, objectACLReq, log, next),
+        ], err => {
+            if (err) {
+                return done(err);
+            }
+            checkObjectMD(keyA, emptyReplicationMD);
+            return done();
+        }));
+
+    it('should not update object metadata if putting a delete marker', done =>
+        async.series([
+            next => putObjectAndCheckMD(keyA, emptyReplicationMD, next),
+            next => objectDelete(authInfo, deleteReq, log, next),
+        ], err => {
+            if (err) {
+                return done(err);
+            }
+            assert.strictEqual(err, null);
+            const objectMD = metadata.keyMaps.get(bucketName).get(keyA);
+            assert.strictEqual(objectMD, undefined, 'Expected object to be ' +
+                `deleted, but found object metadata ${objectMD}`);
+            return done();
+        }));
+
+    describe('Object tagging', () => {
+        beforeEach(done => async.series([
+            next => putObjectAndCheckMD(keyA, emptyReplicationMD, next),
+            next => objectPutTagging(authInfo, taggingPutReq, log, next),
+        ], err => done(err)));
+
+        it('should not update object metadata if putting tag', done => {
+            checkObjectMD(keyA, emptyReplicationMD);
+            return done();
+        });
+
+        it('should not update object metadata if deleting tag', done =>
+            async.series([
+                // Put a new version to update replication MD content array.
+                next => putObjectAndCheckMD(keyA, emptyReplicationMD, next),
+                next => objectDeleteTagging(authInfo, taggingDeleteReq, log,
+                    next),
+            ], err => {
+                if (err) {
+                    return done(err);
+                }
+                checkObjectMD(keyA, emptyReplicationMD);
+                return done();
+            }));
+    });
+});
+
+describe('Replication object MD with bucket replication config', () => {
+    beforeEach(done => {
+        cleanup();
+        async.series([
+            next => bucketPut(authInfo, putBucketReq, log, next),
+            next => bucketPutVersioning(authInfo, versioningReq, log, next),
+            next => bucketPutReplication(authInfo, replicationReq, log, next),
+        ], err => done(err));
+    });
+
+    afterEach(() => cleanup());
+
+    it('should update metadata when replication config prefix applies', done =>
+        putObjectAndCheckMD(keyA, newReplicationMD, done));
+
+    it('should not update metadata when replication config prefix does not ' +
```

Comment:
Good idea. I have added some tests with characters `'acb'` on either side of `${keyA}`. I think the existing test for `${keyB}` should be sufficient with this.